test/functional/legacy/036_regexp_character_classes_spec.lua

```diff
@@ -268,4 +268,15 @@ describe('character classes in regexp', function()
       ABCDEFGHIXYZ
       ABCDEFGHIXYZ]])
   end)
+  it('"\\%1l^#.*" does not match on a line starting with "#". (vim-patch:7.4.1035)', function()
+    source([[
+      1 s/\%#=0\%1l^\t...//g
+      2 s/\%#=1\%2l^\t...//g
+      3 s/\%#=2\%3l^\t...//g
+      4 s/\%#=0\%4l^\t...//g
+      5 s/\%#=1\%5l^\t...//g
+      6 s/\%#=2\%6l^\t...//g]])
+    diff(sixlines(string.sub(punct1, 1)..digits..punct2..upper..punct3..
+      lower..punct4..ctrl2..iso_text))
```

Comment:
Structure of the lua port of this test is hard to read, and longer than the old one... we should try to avoid that in ported tests.

These changes look good though.